doc/docusaurus/docs/delve-deeper/languages.md

```diff
@@ -0,0 +1,102 @@
+---
+sidebar_position: 38
+---
+
+# Overview of Languages Compiling to UPLC
+
+Untyped Plutus Core (UPLC) is the assembly-like language that runs in Cardano nodes for transaction validation.
+The Cardano node ships with a UPLC evaluator, which is a [CEK machine](https://en.wikipedia.org/wiki/CEK_Machine).
+
+UPLC is a low-level programming language, and UPLC programs are as difficult to read as assembly code.
+Besides Plutus Tx, several other high-level languages are designed to target UPLC.
+These languages can be grouped into three categories:
+
+- Standalone DSLs, which are entirely new languages
+- DSLs embedded in existing general-purpose programming languages
+- Subsets of existing general-purpose programming languages
+
+These are also the three common strategies for creating DSLs in general, not limited to blockchains or Cardano.
+Each strategy comes with its own benefits and drawbacks, which we'll discuss next.
+
+## Standalone DSLs
+
+A standalone DSL is a new language with its own syntax and semantics.
+By crafting a new language from scratch, you avoid inheriting the limitations and complexities of existing languages, allowing you to tailor-make it to be as simple, intuitive and elegant as possible to program for the specific domain it targets.
+
+On the other hand, standalone DSLs have some disadvantages.
+First, designing and implementing them can be challenging.
+Not only must the syntax and semantics be created from scratch, but you also need to build the compiler components - such as lexers, parsers, type checkers, desugarers, and optimizers - from the ground up.
```

Comment:
Not very good examples, given that our supposed subset of Haskell has all of those for PIR (and some for TPLC). I think tooling examples would be DE integration, build tools, extensive documentation, community etc.